.github/workflows/milestone-merged-prs.yaml

```
@@ -4,6 +4,8 @@ on:
   pull_request_target:
     types:
       - closed
+    branches:
+      - "main"
 
 jobs:
   milestone_pr:
```

```suggestion
  milestone_pr:
    if: github.head_ref == 'main'
```
Alternative solution.